tests/sentry/integrations/test_pipeline.py

```diff
@@ -89,6 +106,61 @@ def test_with_customer_domain(self, *args):
                 organization_id=self.organization.id, integration_id=integration.id
             ).exists()
 
+    @patch("sentry.signals.integration_added.send_robust")
+    def test_provider_should_check_region_violation(self, *args):
+        """Ensures we validate regions if `provider.is_region_restricted` is set to True"""
+        self.provider.is_region_restricted = True
+        self.pipeline.state.data = {"external_id": self.external_id}
+        with patch(
+            "sentry.integrations.pipeline.is_violating_region_restriction"
+        ) as mock_check_violation:
+            self.pipeline.finish_pipeline()
+            assert mock_check_violation.called
+
+    @patch("sentry.signals.integration_added.send_robust")
+    def test_provider_should_not_check_region_violation(self, *args):
+        """Ensures we don't reject regions if `provider.is_region_restricted` is set to True"""
```

Comment:
```suggestion
        """Ensures we don't reject regions if `provider.is_region_restricted` is set to False"""
```

Isn't this test checking that if region isolation isn't enabled for an integration installation works across regions?